src/dmd/func.d

```diff
@@ -403,7 +403,8 @@ extern (C++) class FuncDeclaration : Declaration
                 return false;
         }
 
-        this.cppnamespace = _scope.namespace;
+        if (_scope)
+            this.cppnamespace = _scope.namespace;
```

Comment:
Just move it in the previous `if`. I assumed `_scope` would be `!is null` because of the check on L388, but I see now that `dsymbolSemantic` sets it to `null`. So the check should either be that function (and probably some other places could be cleaned up), or before the call.